base/strings/search.jl

```diff
@@ -212,20 +206,29 @@ julia> rsearch("aaabbb","b")
 6:6
 ```
 """
-function rsearch(s::AbstractString, c::Chars, i::Integer)
-    e = endof(s)
-    j = search(RevString(s), c, e-i+1)
-    j == 0 && return 0
-    e-j+1
+function rsearch(s::AbstractString, c::Chars, i::Integer=start(s))
+    if isempty(c)
+        return 1 <= i <= nextind(s, endof(s)) ? i : throw(BoundsError(s, i))
```

Comment:
I don't think indices beyond `endof(s)` should be allowed. See also https://github.com/JuliaLang/julia/issues/22548 and https://github.com/JuliaLang/julia/pull/22572. You could probably use `checkbounds` for that? Also, it looks like you could first check the bounds using the block below, and only then handle the special case of empty `c`?

Reply:
`endof(s)` is the last valid index, so `1 <= i <= endof(s)` should be sufficient.

Reply:
Sounds good. I was just mimicking the structure of the `search` code here.

Reply:
Ah, OK. So maybe better keep the same (weird) approach as `search` and fix that later.

Reply:
Seems like I might as well go with the better approach now and update `search` to match.